### PR TITLE
Merge log and metrics interceptor

### PR DESCRIPTION
**Pull request by vrutkovs**
GRPC requests should be handled using one interceptor, which creates a new tracing span and collects request metrics instead of separate procedures

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

See point 1 in https://github.com/cri-o/cri-o/issues/6419. Two different UnaryInterceptors for incoming GRPC requests are created with code duplication

#### Which issue(s) this PR fixes:
Parcially addresses https://github.com/cri-o/cri-o/issues/6419

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Review by saschagrunert — APPROVED** [review]
Code LGTM
/ok-to-test

**Comment by haircommander** [comment]
@vrutkovs could you rebase to fix tests

File: server/container_portforward.go [hunk 1]
```diff
@@ -21,7 +22,7 @@ func (s *Server) PortForward(ctx context.Context, req *types.PortForwardRequest)
 }
 
 func (s StreamService) PortForward(podSandboxID string, port int32, stream io.ReadWriteCloser) error {
-	ctx := log.AddRequestNameAndID(context.Background(), "PortForward")
+	ctx := otel_collector.AddRequestNameAndID(context.Background(), "PortForward")
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply server/container_portforward.go:25]: Good idea, added those too and verified these are showing up in the traces

File: server/container_exec.go [hunk 2]
```diff
@@ -25,7 +26,8 @@ func (s *Server) Exec(ctx context.Context, req *types.ExecRequest) (*types.ExecR
 
 // Exec endpoint for streaming.Runtime
 func (s StreamService) Exec(containerID string, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, resize <-chan remotecommand.TerminalSize) error {
-	ctx, span := log.StartSpan(context.TODO())
+	ctx := otel_collector.AddRequestNameAndID(context.Background(), "PortForward")
```
Comment (haircommander) [inline server/container_exec.go:29]: looks like this should be "Exec"

**Comment by haircommander** [comment]
@sallyom can you PTAL?

File: server/container_exec.go (hunk 2 above)
Comment (vrutkovs) [reply server/container_exec.go]: Good catch, fixed

**Pull request merged**